intera_interface/src/motion_interface/interaction_options.py

```diff
@@ -0,0 +1,244 @@
+            for i in range(0, self.n_dim_joint):
+                self._data.K_nullspace.append(K_nullspace[0])
+        elif len(K_nullspace) < self.n_dim_joint and len(K_nullspace) > 1:
+            rospy.logerr('The number of elements must be n_joint_dim or 1!')
+        elif len(K_nullspace) == self.n_dim_joint:
+            self._data.K_nullspace = deepcopy(K_nullspace)
+        else:
+            rospy.logerr('Invalid input to K_nullspace!')
+
+    def set_force_command(self, force_cmd = None):
+        """
+        @param force_command:
+        --> None:  populate with vector of default values
+        --> [float]:  copy all elements. Checks length.
+        """
+        if force_cmd is None:
+            self._data.force_command =  [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+        elif len(force_cmd) < self.n_dim_cart:
+            rospy.logerr('The number of elements must be 6!')
+        elif len(force_cmd) == self.n_dim_cart:
+            self._data.force_command = deepcopy(force_cmd)
+        else:
+            rospy.logerr('Invalid input to force_command!')
+
+    def set_interaction_frame(self, interaction_frame = None):
+        """
+        @param interaction_frame:
+        --> None:  populate with vector of default values
+        --> otherwise, copy all the elements
+        """
+        if interaction_frame is None:
+            self._data.interaction_frame = Pose()
+            self._data.interaction_frame.position.x = 0.0;
+            self._data.interaction_frame.position.y = 0.0;
+            self._data.interaction_frame.position.z = 0.0;
+            self._data.interaction_frame.orientation.w = 1.0;
+            self._data.interaction_frame.orientation.x = 0.0;
+            self._data.interaction_frame.orientation.y = 0.0;
+            self._data.interaction_frame.orientation.z = 0.0;
+        else:
+            self._data.interaction_frame = interaction_frame
+
+    def set_endpoint_name(self, endpoint_name = None):
+        """
+        @param endpoint_name:
+        --> None:  set it to a default name (right hand)
+        --> string: copy element.
+        """
+        if endpoint_name is None:
+            self.set_endpoint_name('right_hand') # default value
+        else:
+            self._data.endpoint_name = endpoint_name
+
+    def set_in_endpoint_frame(self, in_endframe = None):
+        """
+        @param in_endpoint_frame
+        --> None:  set it to a default frame (False, i.e., base frame)
+        --> bool:  copy element
+        """
+        if in_endframe is None:
+            self.set_in_endpoint_frame(False) # default value
+        else:
+            self._data.in_endpoint_frame = in_endframe
+
+    def set_interaction_control_mode(self, interaction_mode = None):
+        """
+        @param interaction control mode:
+        --> None:  set impedance mode by default
+        --> mode:  set each direction by the input mode (1: impedance, 2: force, 3: impedance with force limit, 4: force with motion limit)
```

Comment:
This line is far too long. Try to limit it to the pep8 recommended 80 characters.